packages/gp2-components/src/templates/PageNotifications.tsx

```diff
@@ -0,0 +1,46 @@
+import { Toast } from '@asap-hub/react-components';
+import {
+  Page,
+  useNotificationContext,
+  Notification,
+} from '@asap-hub/react-context';
+import { css } from '@emotion/react';
+import { ReactNode, useEffect } from 'react';
+
+export type PageNotificationsProps = {
+  page: Page;
+  children: (notification?: Notification) => ReactNode;
+};
+const PageNotifications: React.FC<PageNotificationsProps> = ({
+  page,
+  children,
+}) => {
+  const { notifications, removeNotification } = useNotificationContext();
+  const pageNotifications = notifications.filter(
+    (notification) => notification.page === page,
+  );
+  const displayNotification = pageNotifications[0];
+  useEffect(
+    () => () => displayNotification && removeNotification(displayNotification),
```

Comment:
Should this be run as part of the clean up code?

Reply:
when the components gets dismounted, it cleans up the notification associated to the page,
I thought this was a simple way to dismiss it when we change the route without having to add a listener to the route